### PR TITLE
[firebase.com] Add related domains.

### PR DESCRIPTION
**Pull request by youdly**

### CODE REVIEW

File: src/chrome/content/rules/Firebase.xml [hunk 1]
```diff
@@ -1,20 +1,37 @@
...
+		<test url="http://s-dal5-nss-19.firebaseio.com/" />
+		<test url="http://s-dal5-nss-46.firebaseio.com/" />
+		<test url="http://s-softlayer.firebaseio.com/" />
+		<test url="http://www.firebaseio.com/" />
 
 	<securecookie host="^auth\.firebase\.com$" name=".+" />
+	<securecookie host="^.*\.firebaseapp\.com$" name=".+" />
+	<securecookie host="^.*\.firebaseio\.com$" name=".+" />
 
+	<!-- could not resolve host -->
```
Comment (J0WI) [inline src/chrome/content/rules/Firebase.xml]: We don't redirect hosts that doesn't exist.

**Comment by youdly** [comment]
Removed redirect.

File: src/chrome/content/rules/Firebase.xml [hunk 2]
```diff
@@ -1,20 +1,33 @@
...
-	<target host="firebase.com" />
+	<target host=     "firebase.com" />
 	<target host="auth.firebase.com" />
-	<target host="www.firebase.com" />
-
+	<target host= "cdn.firebase.com" />
+	<target host= "www.firebase.com" />
+	<target host=     "firebaseapp.com" />
+	<target host=   "*.firebaseapp.com" />
+	<target host=     "firebaseio.com" />
```
Comment (J0WI) [inline src/chrome/content/rules/Firebase.xml]: You should remove this non existent host

**Comment by youdly** [comment]
Removed `firebaseio.com`.

**Comment by jeremyn** [comment]
@youdly Thanks for the pull request but it needs some changes. Are you still interested in working on this?

**Comment by youdly** [comment]
@jeremyn Any suggestion for the changes?

**Comment by jeremyn** [comment]
@youdly I don't want to be too specific because I'm not actually reviewing this and don't want you to feel cheated if you make a bunch of changes and I don't merge it. However some things I notice are: we don't like wildcards; the sorting is off; the formatting is off. Please take a look at [`CONTRIBUTING.md`](https://github.com/EFForg/https-everywhere/blob/master/CONTRIBUTING.md).

**Comment by youdly** [comment]
Fix sorting and formatting. I try to avoid wildcard, but it is necessary in this ruleset because there are too many subdomains to be practically included.

File: src/chrome/content/rules/Firebase.xml [hunk 3]
```diff
@@ -1,20 +1,31 @@
...
-				-->
 	<target host="firebase.com" />
-	<target host="auth.firebase.com" />
 	<target host="www.firebase.com" />
-
+	<target host="auth.firebase.com" />
+	<target host="cdn.firebase.com" />
+	<target host="firebaseapp.com" />
+	<target host="*.firebaseapp.com" />
+	<target host="*.firebaseio.com" />
```
Comment (J0WI) [inline src/chrome/content/rules/Firebase.xml]: Please add comment about the wildcard DNS entry for both wildcard targets.

File: src/chrome/content/rules/Firebase.xml [hunk 4]
```diff
@@ -1,20 +1,31 @@
 <!--
 	Nonfunctional subdomains:
 
-		- status
+		- status		(timeout)
```
Comment (J0WI) [inline src/chrome/content/rules/Firebase.xml]: Since this ruleset has multiple domains you should write the full domain here.

**Comment by J0WI** [comment]
ping @youdly

**Comment by youdly** [comment]
Added comment

**Comment by J0WI** [comment]
Thanks!

**Pull request merged**